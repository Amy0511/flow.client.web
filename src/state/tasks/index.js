/* eslint-disable no-console*/
import createReducer from "@boomerang/boomerang-utilities/lib/createReducer";
import requestGenerator from "@boomerang/boomerang-utilities/lib/requestGenerator";

// Action types
export const types = {
  RESET_TASKS: "RESET_TASKS",
  FETCH_TASKS_REQUEST: "FETCH_TASKS_REQUEST",
  FETCH_TASKS_SUCCESS: "FETCH_TASKS_SUCCESS",
  FETCH_TASKS_FAILURE: "FETCH_TASKS_FAILURE"
};
Object.freeze(types);

// Initial state
export const initialState = {
  isFetching: false,
  status: "",
  error: "",
  data: []
};

//action handlers
const actionHandlers = {
  [types.RESET_TASKS]: () => {
    return { ...initialState };
  },
  [types.FETCH_TASKS_REQUEST]: state => {
    return { ...state, isFetching: true };
  },
  [types.FETCH_TASKS_SUCCESS]: (state, action) => {
    return { ...state, isFetching: false, status: "success", data: action.data };
  },
  [types.FETCH_TASKS_FAILURE]: (state, action) => {
    return { ...state, isFetching: false, status: "failure", error: action.error };
  }
};

//reducer exported by default
export default createReducer(initialState, actionHandlers);

/*
 action creators declared to be passed into the request generator boilerplate
*/
const resetFetchTasks = () => ({ type: types.RESET_TASKS });
const fetchTasksRequest = () => ({ type: types.FETCH_TASKS_REQUEST });
const fetchTasksSuccess = data => ({ type: types.FETCH_TASKS_SUCCESS, data });
const fetchTasksFailure = error => ({ type: types.FETCH_TASKS_FAILURE, error });

const fetchTasksCreators = {
  reset: resetFetchTasks,
  request: fetchTasksRequest,
  success: fetchTasksSuccess,
  failure: fetchTasksFailure
};

const fetchTasksApi = requestGenerator(fetchTasksCreators);

const fetchTasks = url => dispatch => dispatch(fetchTasksApi.request({ method: "get", url }));

const cancelFetchTasks = () => dispatch => dispatch(fetchTasksApi.cancelRequest());

//actions
export const actions = {
  resetFetchTasks,
  fetchTasksRequest,
  fetchTasksSuccess,
  fetchTasksFailure,
  fetchTasks,
  cancelFetchTasks
};
