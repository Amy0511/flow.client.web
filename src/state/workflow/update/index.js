/* eslint-disable no-console*/
import createReducer from "@boomerang/boomerang-utilities/lib/createReducer";
import requestGenerator from "@boomerang/boomerang-utilities/lib/requestGenerator";

// Action types
export const types = {
  UPDATE_WORKFLOW_SUCCESS: "UPDATE_WORKFLOW_SUCCESS",
  UPDATE_WORKFLOW_FAILURE: "UPDATE_WORKFLOW_FAILURE",
  UPDATE_WORKFLOW_REQUEST: "UPDATE_WORKFLOW_REQUEST",
  UPDATE_WORKFLOW_RESET: "UPDATE_WORKFLOW_RESET"
};
Object.freeze(types);

//initial state
export const initialState = {
  isUpdating: false,
  status: "",
  error: "",
  data: []
};

//action handlers
const actionHandlers = {
  [types.UPDATE_WORKFLOW_SUCCESS]: (state, action) => ({
    ...state,
    isUpdating: false,
    status: "success",
    data: action.response
  }),
  [types.UPDATE_WORKFLOW_FAILURE]: (state, action) => ({
    ...state,
    isUpdating: false,
    status: "failure",
    data: [],
    error: action.error
  }),
  [types.UPDATE_WORKFLOW_REQUEST]: state => ({ ...state, isUpdating: true, status: "" }),
  [types.UPDATE_WORKFLOW_RESET]: () => ({ ...initialState })
};

export default createReducer(initialState, actionHandlers);

/*
 action creators declared to be passed into the GET request generator boilerplate
*/

const updateRequest = () => ({ type: types.UPDATE_WORKFLOW_REQUEST });
const updateSuccess = data => ({ type: types.UPDATE_WORKFLOW_SUCCESS, data });
const updateFailure = error => ({ type: types.UPDATE_WORKFLOW_FAILURE, error });
const reset = () => ({ type: types.UPDATE_WORKFLOW_RESET });

const updateActionCreators = {
  reset: reset,
  request: updateRequest,
  success: updateSuccess,
  failure: updateFailure
};

const updateApi = requestGenerator(updateActionCreators);
const update = (url, data) => dispatch => dispatch(updateApi.request({ method: "put", url, data }));
const cancel = () => dispatch => dispatch(updateApi.cancelRequest());

/*
 action creators declared to be passed into the GET request generator boilerplate
*/

//actions
export const actions = {
  update,
  updateRequest,
  updateFailure,
  updateSuccess,
  cancel,
  reset
};
