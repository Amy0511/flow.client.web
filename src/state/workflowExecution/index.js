/* eslint-disable no-console*/
import createReducer from "@boomerang/boomerang-utilities/lib/createReducer";
import requestGenerator from "@boomerang/boomerang-utilities/lib/requestGenerator";

// Action types
export const types = {
  WORKFLOW_EXECUTION_RESET: "WORKFLOW_EXECUTION_RESET",
  FETCH_WORKFLOW_EXECUTION_REQUEST: "FETCH_WORKFLOW_EXECUTION_REQUEST",
  FETCH_WORKFLOW_EXECUTION_SUCCESS: "FETCH_WORKFLOW_EXECUTION_SUCCESS",
  FETCH_WORKFLOW_EXECUTION_FAILURE: "FETCH_WORKFLOW_EXECUTION_FAILURE"
};
Object.freeze(types);

//initial state
export const initialState = {
  isFetching: false,
  status: "",
  error: "",
  data: {}
};

//action handlers
const actionHandlers = {
  [types.WORKFLOW_EXECUTION_RESET]: () => {
    return { ...initialState };
  },
  [types.FETCH_WORKFLOW_EXECUTION_REQUEST]: state => {
    return { ...state, isFetching: true };
  },
  [types.FETCH_WORKFLOW_EXECUTION_SUCCESS]: (state, action) => {
    return { ...state, isFetching: false, status: "success", data: action.data };
  },
  [types.FETCH_WORKFLOW_EXECUTION_FAILURE]: (state, action) => {
    return { ...state, isFetching: false, status: "failure", error: action.error };
  }
};

export default createReducer(initialState, actionHandlers);

/*
 action creators declared to be passed into the GET request generator boilerplate
*/
const fetchRequest = () => ({ type: types.FETCH_WORKFLOW_EXECUTION_REQUEST });
const fetchSuccess = data => ({ type: types.FETCH_WORKFLOW_EXECUTION_SUCCESS, data });
const fetchFailure = error => ({ type: types.FETCH_WORKFLOW_EXECUTION_FAILURE, error });
const reset = () => ({ type: types.WORKFLOW_EXECUTION_RESET });

const fetchActionCreators = {
  reset: reset,
  request: fetchRequest,
  success: fetchSuccess,
  failure: fetchFailure
};

const fetchApi = requestGenerator(fetchActionCreators);
const fetch = url => dispatch => dispatch(fetchApi.request({ method: "get", url }));
const cancel = () => dispatch => dispatch(fetchApi.cancelRequest());

/*
 action creators declared to be passed into the GET request generator boilerplate
*/

//actions
export const actions = {
  fetch,
  fetchRequest,
  fetchFailure,
  fetchSuccess,
  cancel,
  reset
};
