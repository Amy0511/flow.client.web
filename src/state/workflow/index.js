import { combineReducers } from "redux";
import fetch from "./fetch";
import update from "./update";

const rootReducer = combineReducers({
  fetch,
  update
});

export default rootReducer;
