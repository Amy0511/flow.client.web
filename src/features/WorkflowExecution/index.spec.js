import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { actions as tasksActions } from "State/tasks";
import { actions as workflowActions } from "State/workflow";
import { actions as workflowExecutionActions } from "State/workflowExecution";
import { actions as workflowExecutionActiveNodeActions } from "State/workflowExecutionActiveNode";
import { actions as workflowRevisionActions } from "State/workflowRevision";
import { WorkflowExecutionContainer } from "./index";

jest.mock("./Main", () => "Main");

const match = { params: {} };

const tasks = {
  isFetching: false,
  status: "",
  error: "",
  data: []
};

const workflow = {
  isFetching: false,
  isUpdating: false,
  isCreating: false,
  fetchingStatus: "",
  updatingStatus: "",
  creatingStatus: "",
  error: "",
  data: {}
};

const workflowExecution = {
  isFetching: false,
  status: "",
  error: "",
  data: {}
};

const workflowExecutionActiveNode = {
  activeNode: {}
};

const workflowRevision = {
  isFetching: false,
  isCreating: false,
  fetchingStatus: "",
  creatingStatus: "",
  error: "",
  dag: undefined,
  version: "",
  config: {}
};

describe("WorkflowExecutionContainer --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowExecutionContainer", () => {
    const renderedValue = renderer.create(
      <WorkflowExecutionContainer
        match={match}
        tasks={tasks}
        tasksActions={tasksActions}
        workflow={workflow}
        workflowActions={workflowActions}
        workflowExecution={workflowExecution}
        workflowExecutionActions={workflowExecutionActions}
        workflowExecutionActiveNode={workflowExecutionActiveNode}
        workflowExecutionActiveNodeActions={workflowExecutionActiveNodeActions}
        workflowRevision={workflowRevision}
        workflowRevisionActions={workflowRevisionActions}
      />
    );
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("WorkflowExecutionContainer --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <WorkflowExecutionContainer
        match={match}
        tasks={tasks}
        tasksActions={tasksActions}
        workflow={workflow}
        workflowActions={workflowActions}
        workflowExecution={workflowExecution}
        workflowExecutionActions={workflowExecutionActions}
        workflowExecutionActiveNode={workflowExecutionActiveNode}
        workflowExecutionActiveNodeActions={workflowExecutionActiveNodeActions}
        workflowRevision={workflowRevision}
        workflowRevisionActions={workflowRevisionActions}
      />
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
