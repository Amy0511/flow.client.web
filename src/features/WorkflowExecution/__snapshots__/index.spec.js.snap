// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowExecutionContainer --- Snapshot Capturing Snapshot of WorkflowExecutionContainer 1`] = `null`;
